app/controllers/stories_controller.rb

```diff
@@ -50,6 +51,10 @@ def show
 
   private
 
+  def set_admin
+    @is_admin = current_user&.any_admin?
```

Comment:
Is this even legal to do this? It's not allowed to access `current_user` after `set_cache_control_headers`, so..

Reply:
Yeah. We can't set this after `set_cache_control_headers` or else it will effectively be cached for the next user.

I think we can only set it in show contexts after we have determined that the article is spam and we have disabled cache headers, as per my other comment.